bigquery/tests/unit/test_client.py

```diff
@@ -5161,3 +5161,84 @@ def test__do_multipart_upload_wrong_size(self):
 
         with pytest.raises(ValueError):
             client._do_multipart_upload(file_obj, {}, file_obj_len + 1, None)
+
+    def test__schema_from_json(self):
+        from google.cloud.bigquery.schema import SchemaField
+
+        file_content = """[
+          {
+            "description": "quarter",
+            "mode": "REQUIRED",
+            "name": "qtr",
+            "type": "STRING"
+          },
+          {
+            "description": "sales representative",
+            "mode": "NULLABLE",
+            "name": "rep",
+            "type": "STRING"
+          },
+          {
+            "description": "total sales",
+            "mode": "NULLABLE",
+            "name": "sales",
+            "type": "FLOAT"
+          }
+        ]"""
+        expected = list()
+        json_data = json.loads(file_content)
```

Comment:
You're basically repeating the function definition here. That's not ideal. You've basically written a [change-detector test](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html). I'd prefer to see actual `schema.SchemaField` constructors in tests.

I do like that you've mocked out `open` and the file contents, so keep that. Just change how you construct `expected`.

Reply:
Made changes to include schema.SchemaField as requested.